File: Quest/Assets/Src/Cards/Ally Cards/SirGawain.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class SirGawain : AllyCard
{

    public SirGawain() : base("Sir Gawain", 10, 1) //base 10
    {

    }

    public override int getBattlePoints(GameState gState) //override 20
    {
        if (gState.getCurrentStoryCard().getName().Equals("Test of the Green Knight"))
        {
            return battlePoints = 20;
        }
        return battlePoints = 10;
    }

    public override Ability[] GetAbilities() //no ability
    {
        return null;
    }

}

File: eclipse/Quest/src/model/cards/allyCards/SirGawain.java
package model.cards.allyCards;

import model.cards.AllyCard;
import model.game.GameState;

public class SirGawain extends AllyCard {

	public SirGawain() {
		super("Sir Gawain", 10, 0); //base 10
	}

    public int getBattlePoints(GameState gState) //override 20
    {
        if (gState.getCurrentStoryCard().getName().equals("Test of the Green Knight"))
        {
            return battlePoints = 20;
        }
        return battlePoints = 10;
    }
}
